ee/clickhouse/queries/funnels/funnel_correlation_persons.py

```diff
@@ -1,144 +1,189 @@
         """
+
         params = {
             **funnel_persons_params,
-            **person_query_params,
+            **prop_params,
+            "target_event": self._filter.correlation_person_entity.id,
+            "funnel_step_names": [entity.id for entity in self._filter.events],
             "target_step": len(self._filter.entities),
         }
 
         return query, params
 
-    def get_events_query(self) -> Tuple[str, Dict[str, Any]]:
 
-        if not self._filter.correlation_person_entity:
-            raise ValidationError("No entity for persons specified")
+class _FunnelPropertyCorrelationActors(ActorBaseQuery):
+    def __init__(self, filter: Filter, team: Team, base_uri: str = "/") -> None:
+        self._funnel_correlation = FunnelCorrelation(filter, team, base_uri=base_uri)
+        super().__init__(team, filter)
 
-        assert isinstance(self._filter.correlation_person_entity, Entity)
+    @cached_property
+    def is_aggregating_by_groups(self) -> bool:
+        return self._filter.aggregation_group_type_index is not None
 
-        funnel_persons_query, funnel_persons_params = self._funnel_correlation.get_funnel_persons_cte()
+    def actor_query(self, extra_fields: Optional[List[str]] = None):
+        if not self._filter.correlation_property_values:
+            raise ValidationError("Property Correlation expects atleast one Property to get persons for")
 
-        prop_filters = self._filter.correlation_person_entity.properties
-        prop_query, prop_params = FunnelEventQuery(self._filter, self._team.pk)._get_props(prop_filters)
+        funnel_persons_query, funnel_persons_params = self._funnel_correlation.get_funnel_persons_cte()
 
         conversion_filter = (
-            f'AND person.steps {"=" if self._filter.correlation_persons_converted else "<>"} target_step'
+            f'funnel_people.steps {"=" if self._filter.correlation_persons_converted else "<>"} target_step'
             if self._filter.correlation_persons_converted is not None
             else ""
         )
 
-        event_join_query = self._funnel_correlation._get_events_join_query()
+        actor_join_subquery, actor_join_subquery_params = self._get_actor_subquery()
+        group_filters, group_filters_params = self._get_group_filters()
 
         query = f"""
             WITH
                 funnel_people as ({funnel_persons_query}),
-                toDateTime(%(date_to)s) AS date_to,
-                toDateTime(%(date_from)s) AS date_from,
-                %(target_step)s AS target_step,
-                %(funnel_step_names)s as funnel_step_names
+                %(target_step)s AS target_step
             SELECT
-                DISTINCT person.person_id as person_id
-            FROM events AS event
-                {event_join_query}
-                AND event.event = %(target_event)s
-                {conversion_filter}
-                {prop_query}
+                DISTINCT funnel_people.person_id as person_id
```

Comment:
While funnel_people needs to return data as person_id as a kludge we should solve, no reason we shouldn't `AS actor_id` here.